File: 0360 322 Object-Oriented Software Analysis and Design/Assignments/A3/src/LandsOnVisitor.java
public class LandsOnVisitor extends EntityVisitor {

    // second argument
    private Entity secondEntity;

    LandsOnVisitor(Entity secondEntity) {               // construct, remember second argument
        this.secondEntity = secondEntity;
    }

    public void visit(Asteroid a) {                     // asteroid lands on a ...??
        secondEntity.doLand(a);                         // determine what secondEntity is
    }

    public void visit(Spaceship s) {                    // spaceship lands on a ...??
        secondEntity.doLand(s);                         // determine what secondEntity is
    }

    public void visit(Moon m) {                         // moon lands on a ...??
        secondEntity.doLand(m);                         // determine what secondEntity is
    }
}

File: 0360 322 Object-Oriented Software Analysis and Design/Assignments/DynamicDispatch/DynamicDispatch/LandsOnVisitor.cs
using System;


namespace A3Test
{
    public class LandsOnVisitor : IVisitor
    {
        private readonly SpaceEntity _secondEntity;

        public LandsOnVisitor(SpaceEntity secondEntity) => _secondEntity = secondEntity;

        public void Visit(Asteroid astereoid) => Interaction.Land(_secondEntity, astereoid);

        public void Visit(Spaceship spaceship) => Interaction.Land(_secondEntity, spaceship);

        public void Visit(Moon moon) => Interaction.Land(_secondEntity, moon);
    }
}
